src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,58 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import './FindMovie.scss';
+import classNames from 'classnames';
+import { MovieCard } from '../MovieCard';
+
+import { Movie } from '../../types/Movie';
+
+interface Props {
+  query: string;
+  setQuery: (query: string) => void;
+  movie: Movie | null;
+  setMovie: (movie: Movie | null) => void;
+  movies: Movie[];
+  setMovies: (movies: Movie[]) => void;
+  isSubmitted: boolean;
+  setIsSubmitted: (isFetching: boolean) => void;
+  showError: boolean;
+  setShowError: (showError: boolean) => void;
+}
+
+export const FindMovie: React.FC<Props> = ({
+  query, setQuery, movie, setMovies, movies, setIsSubmitted, isSubmitted,
+  setMovie, showError, setShowError,
+
+}) => {
+  useEffect(() => {
+    setShowError(false);
+  }, [query]);
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    setIsSubmitted(true);
+  };
+
+  const handleAdd = () => {
+    if (movie) {
+      setMovies([...movies, movie]);
+
+      if (movies.length
+        && movies.some(({ imdbId }) => movie.imdbId === imdbId)) {
+        setMovies(movies);
+      }
+    }
+
+    setQuery('');
+    setMovie(null);
+  };
```

Comment:
You can move this function to App.tsx to not pass such a lot of props to this component. But it's up to you.